src/quantum/azext_quantum/operations/job.py

```diff
@@ -174,8 +194,202 @@ def _has_completed(job):
 
 
 def submit(cmd, program_args, resource_group_name, workspace_name, location, target_id,
-           project=None, job_name=None, shots=None, storage=None, no_build=False, job_params=None,
-           target_capability=None):
+           project=None, job_name=None, shots=None, storage=None, no_build=False, job_params=None, target_capability=None,
+           job_input_file=None, job_input_format=None, job_output_format=None, entry_point=None):
+    """
+    Submit a quantum program to run on Azure Quantum.
+    """
+    if job_input_format is None:
+        return _submit_qsharp(cmd, program_args, resource_group_name, workspace_name, location, target_id,
+                              project, job_name, shots, storage, no_build, job_params, target_capability)
+
+    return _submit_directly_to_service(cmd, resource_group_name, workspace_name, location, target_id,
+                                       job_name, shots, storage, job_params, target_capability,
+                                       job_input_file, job_input_format, job_output_format, entry_point)
+
+
+def _submit_directly_to_service(cmd, resource_group_name, workspace_name, location, target_id,
+                                job_name, shots, storage, job_params, target_capability,
+                                job_input_file, job_input_format, job_output_format, entry_point):
+    """
+    Submit QIR bitcode, QIO problem JSON, or a pass-through job to run on Azure Quantum.
+    """
+
+    # Get workspace and target information
+    ws_info = WorkspaceInfo(cmd, resource_group_name, workspace_name, location)
+    if ws_info is None:
+        raise AzureInternalError("Failed to get workspace information.")
+    target_info = TargetInfo(cmd, target_id)
+    if target_info is None:
+        raise AzureInternalError("Failed to get target information.")
+    provider_id = target_info.target_id.split('.')[0]                   # <<<<< Is provider_id case sensitive? <<<<<
+
+    # Microsoft provider_id values don't follow the same pattern as all the other companies' provider_id values:
+    # A first-party provider ID might might have a dot in it.
+    if provider_id.lower() == "microsoft":
+        after_the_dot = target_info.target_id.split('.')[1]
+        if after_the_dot.lower() == "simulator" or after_the_dot.lower() == "fleetmanagement":  # <<<<< Are there more names like these?
+            provider_id = f"{provider_id}.{after_the_dot}"
+
+    # Identify the type of job being submitted
+    lc_job_input_format = job_input_format.lower()
+    if "qir.v" in lc_job_input_format:       # QCI uses "qir.v1", but Rigetti uses rigetti.qir.v1  <<<<< What about Quantinuum?
+        job_type = QIR_JOB
+    elif lc_job_input_format == "microsoft.qio.v2":
+        job_type = QIO_JOB
+    else:
+        job_type = PASS_THROUGH_JOB
+
+    # If output format is not specified, supply a default for QIR or QIO jobs   # <<<<< Is this out-of-scope?
```

Comment:
I like the idea of defaulting output format. Looks like microsoft.quantum-results.v1 is standard. Check with Cesar to confirm

Reply:
Done.